hepcrawl/parsers/elsevier.py

```diff
@@ -0,0 +1,615 @@
+            self.builder.add_author(author)
+        self.builder.add_publication_info(**self.publication_info)
+        for collab in self.collaborations:
+            self.builder.add_collaboration(collab)
+        for doi in self.dois:
+            self.builder.add_doi(**doi)
+        for keyword in self.keywords:
+            self.builder.add_keyword(keyword)
+        self.builder.add_imprint_date(self.publication_date.dumps())
+        for reference in self.references:
+            self.builder.add_reference(reference)
+
+        return self.builder.record
+
+    @property
+    def references(self):
+        """Extract a Elsevier record into an Inspire HEP references record.
+
+        Returns:
+            List[dict]: an array of reference schema records, representing
+                the references in the record
+        """
+        ref_nodes = self.root.xpath('.//bib-reference')
+        return list(
+            itertools.chain.from_iterable(
+                self.get_reference_iter(node) for node in ref_nodes
+            )
+        )
+
+
+    remove_tags_config_abstract = {
+        'allowed_tags': ['sup', 'sub'],
+        'allowed_trees': ['math'],
+        'strip': 'self::pub-id|self::issn'
+    }
+
+    @property
+    def abstract(self):
+        abstract_nodes = self.root.xpath(".//head/abstract/abstract-sec/simple-para")
+
+        if not abstract_nodes:
+            return
+
+        abstract = remove_tags(abstract_nodes[0], **self.remove_tags_config_abstract).strip('/ /n')
+        return abstract
+
+    @property
+    def article_type(self):
+        """Return a doctype mapped from abbreviation."""
+        abbrv_doctype = self.root.xpath(".//@docsubtype").extract()
+        doctype = ''
+        if abbrv_doctype:
+            doctype = DOCTYPE_MAPPING[get_first(abbrv_doctype)]
+        elif self.top_level_document_type in ['article', 'simple-article']:
+            doctype = 'article'
+        elif self.top_level_document_type in ['book', 'book-chapter', 'book-review', 'simple-book']:
+            doctype = "book"
+        elif self.top_level_document_type == 'exam':
+            doctype = 'exam'
+        if self.root.xpath("./conference-info"):
+            doctype = "conference paper"
+        if doctype:
+            return doctype
+
+    @property
+    def top_level_document_type(self):
+        possible_article_types = 'self::article or self::simple-article'
+        possible_book_types = 'self::book or self::book-chapter or self::book-review or self::simple-book'
+        possible_top_level_document_types = "./*[{0} or {1} or self::exam]".format(
+            possible_article_types, possible_book_types
+        )
+        top_level_document_type_node = self.root.xpath(possible_top_level_document_types)
+        if top_level_document_type_node:
+            top_level_document_type = top_level_document_type_node[0].xpath('name()').extract_first()
+            return top_level_document_type
```

Comment:
I don't think you have to do this. If you know that the node selectors are at the n-th level, using something like `./*/foo/bar` is perfectly fine. As long as you don't start all your selectors with `//`, it's ok.